src/foam/nanos/approval/Approvable.js

```diff
@@ -120,7 +120,26 @@ foam.CLASS({
             ? `${modelString}: ${obj.toSummary()}`
             :  `(${modelString}:${this.objId}) UPDATE`
         });
-      }
+      },
+      javaCode: `
+        String modelString = getDaoKey();
+
+        modelString = modelString.replaceAll("local","");
+        modelString = modelString.replaceAll("DAO","");
+        modelString = '(' + modelString + ':' + getObjId().toString() + ") UPDATE";
```

Comment:
StringBuilder would be appropriate, considering how often this is invoked.

Reply:
Done.